storage/series/series.go

```diff
@@ -319,7 +319,7 @@ func (s *dbSeries) FetchBlocks(
 		id:         s.id,
 		retriever:  s.blockRetriever,
 		onRetrieve: s.onRetrieveBlock,
-	}.fetchBlocksWithBlocksMapAndBuffer(ctx, starts, s.blocks, s.buffer)
+	}.fetchBlocksWithBlocksMapAndBuffer(ctx, starts, time.Duration(0), s.blocks, s.buffer)
```

Comment:
This wouldn't work right?  Otherwise end time will always be start time.  As per previous comment though, in `storage/series/reader.go` instead of passing the block duration / block size here,  let's just make the underlying blocks and block retriever return a `xio.BlockReader`.

Reply:
My bad, I thought in this path, where seriesBlocks was not nil, it would use the `stream.End()` values for each block, but if it were nil here or if there weren't enough blocks passed through it would fail